File: Problems/Leetcode/3001-4000/3101-3200/3191.cpp
#include <bits/stdc++.h>

using namespace std;

static constexpr int MOD = 1e9 + 7;

class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();
        int cnt = 0;
        for (int i = 0; i < n - 2; i++) {
            if (nums[i] == 1) {
                continue;
            }

            cnt++;
            nums[i] ^= 1;
            nums[i + 1] ^= 1;
            nums[i + 2] ^= 1;
        }

        return nums[n - 3] + nums[n - 2] + nums[n - 1] == 3 ? cnt : -1;
    }
};

int main() {
    Solution sol;

    vector<int> nums {0, 1, 1, 1, 0, 0};
    sol.minOperations(nums);

    return 0;
}

File: Problems/Leetcode/3001-4000/3101-3200/3191.py
from typing import List

"""
https://codeforces.com/problemset/problem/1955/E
"""


class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = 0
        for i in range(n - 2):
            if nums[i] == 0:
                nums[i + 1] ^= 1
                nums[i + 2] ^= 1
                cnt += 1
        return cnt if nums[n - 1] == 1 and nums[n - 2] == 1 else -1
